### DISCUSSION TITLE
Error during installation

### DISCUSSION THREAD

**Discussion by HAMMERGAMES01** [itzg/docker-minecraft-server#3182] [Q&A]
Hardware -> 4Cores ARM, 24GB RAM
OS -> Ubuntu 24.04

Hosted via coolify ( https://coolify.io/ )

I keep having these two errors:
1. [ERROR] Failed to auto-install CurseForge modpack
2. ERROR : Invalid parameter provided for 'install-curseforge' command: Access to https://api.curseforge.com is forbidden or rate-limit has been exceeded. Ensure CF_API_KEY is set to a valid API key from https://console.curseforge.com/ or allow rate-limit to reset.

My docker file looks like this:
![{C8FE83C2-2617-4134-ADC7-EE95454BF3EE}](https://github.com/user-attachments/assets/194fdfb5-6fbe-4962-b5c8-b72b382a5a1c)

MODPACK_PLATFORM=AUTO_CURSEFORGE
CF_API_KEY= private but is like this $11$22$33aaaaaaaaaaaaaaaaaaaaaaaaaa
CF_PAGE_URL=https://www.curseforge.com/minecraft/modpacks/boosted-fps
MINECRAFT_MEMORY=8G

Surely I'm doing something wrong but I don't understand what

**Comment by itzg** [comment]
Looks like you haven't escaped the dollar signs in the api key

https://docker-minecraft-server.readthedocs.io/en/latest/types-and-platforms/mod-platforms/auto-curseforge/#api-key

**Reply by HAMMERGAMES01** [reply]
Now my CF_API_KEY look like this $$11$22$33aaaaaaaaaaaaaaaaaaaaaaaaaa but i still have this errors
![{35D0AE9B-10D2-42F8-883E-9D7145E3D99D}](https://github.com/user-attachments/assets/74a601e3-f593-42b7-8a76-4bff069a8f63)

**Reply by HAMMERGAMES01** [reply]
> Looks like you haven't escaped the dollar signs in the api key
> 
> https://docker-minecraft-server.readthedocs.io/en/latest/types-and-platforms/mod-platforms/auto-curseforge/#api-key

I tried to take an example already made but even there I have the same problem, now I show the private key and then regenerate it but at least maybe so you can better understand where I'm wrong

I used this example: https://github.com/itzg/docker-minecraft-server/blob/master/examples/bettermc/docker-compose.yml

Docker file:
![{65A32B8D-D93E-4391-82B4-8A1BFFB2B018}](https://github.com/user-attachments/assets/d45ec791-d77e-4f65-9242-d3f3b2eea0ee)

Key:
![{C1C25665-E623-46EA-81C5-F47FCFEA5820}](https://github.com/user-attachments/assets/d1048487-b4c7-4ba5-93cc-d4a9ccf06fe2)

Errors:
![{3A73F74F-1510-4793-89B2-FC3FD828ECDA}](https://github.com/user-attachments/assets/dddf12a4-bf7f-40fe-a742-852e263f5db3)

**Reply by itzg** [reply]
Escape **all** dollar signs. All three of them. Please refer to the docs again.

**Reply by HAMMERGAMES01** [reply]
> Escape **all** dollar signs. All three of them. Please refer to the docs again.

I wrote the message wrong if you look at the photo there are $$ in all three places.

![image](https://github.com/user-attachments/assets/3d2fe7e8-de62-4ced-b352-663a98ee3b38)

**Reply by itzg** [reply]
I'm confused how that UI is involved with a compose file. If it shows two dollar signs there, then I'm a bit worried. The escaped dollar signs is only for compose file and shell environment handling. 

CurseForge never has documented their rate limit reset interval. Maybe wait an hour or more before starting the container again. Also, they probably rate limit by IP address, so make sure there aren't any other instances crash looping at your location. 

Otherwise, I'm out of ideas.

**Reply by JorenRothman** [reply]
I tried recreated this issue and the issue is with Coolify, it does something weird with the environmental variable. I tested the same configuration in Dokploy (another docker Web UI) and that worked straight away.

**Comment by thKali** [comment]
some workarround?

**Comment by thKali** [comment]
using https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.418

in edit docker compose

keep "escape especial characters" option ticked
![image](https://github.com/user-attachments/assets/1a55992b-640b-4059-819c-200558e0242b)

in env variables
tick "literal"
![image](https://github.com/user-attachments/assets/1cc72d38-64a1-4937-a054-a0026d06a997)

on your api key

$2a$10$OwsJ/wxXKIydkfHg
add scapes to "$"
\\$2a\\$10\\$OwsJ/wxXKIy

special thanks to @datner  and @atikospeed from the coolLabs discord
https://discord.com/channels/459365938081431553/1345366136346382418/1375588431899394291

**Reply by AtikoSpeed** [reply]
Oh my god someone appreciates me?

**Reply by datner** [reply]
@datner * :)

**Reply by thKali** [reply]
is your discord user ; - ;

fixed
